### ISSUE TITLE
View going blank after few data refreshes

### ISSUE THREAD

**Issue by ram992** [siralam/LoopingViewPager#7]
Hello @siralam, I am facing a very strange issue. I am using this library for my ViewPager.
The Architecture is like this : I am using ViewPagerIndicator in my MainActivity which has 5 fragments. In the first fragment (this has a recyclerView) the first item is a viewPager, and I am using this library for that. I have the same setup as described in the Git. 
I am also providing swipe down to refresh in the app and after few refreshes, in the first page the ViewPager is going blank(Doesn't display anything but the clicks work). If I scroll down a bit it displays the images(I think the recyclerView in the FirstFragment is redrawing its children at this point).  
How can I fix it.

**Comment by siralam** [comment]
This also looks strange to me, at this point I have too little information to help.
Are you sure you have specified `viewpagerAspectRatio` in the xml?

If yes, would you mind provide a sample project which is able to reproduce the problem? If you don't want to clone one, and if you don't mind, you can also share the project with me privately.

**Comment by ram992** [comment]
I cannot share the project, but will try to provide you a sample in two days, you can also try if you have time,
just put the swipe down to refresh logic in the demo that is already there, and refresh the layout.

**Comment by siralam** [comment]
Just by adding a `SwipeRefreshLayout` and performing change dataset function (Which is the same as the what the button does) cannot reproduce the problem.

How do you "refresh the layout" ?

**Comment by ram992** [comment]
I have an Adapter for inflating different layouts in the ReyclerView of the FirstFragment basing on layoutType
Once user swipes down, I simply make an API call set the data in the Fragment's RecyclerView and that triggers each components logic. In ViewPager I just check whether I have valid data or not before passing to Adapter(child of LoopingPagerAdapter). If the data is invalid, then I will make an API call to fetch it(while doing this I simple load dummyData with a placeHolderImage). Once data is recieved I will call notifyDataSetCHanged() on the fragment's recyclerView that will make the ViewPager call it's adapter, since the data is valid now, it will be applied.

**Comment by siralam** [comment]
I guess the problem lies on how to refresh the data but not the viewpager. Looking forward to your sample project which demonstrate the problem!

**Comment by ram992** [comment]
Sure, will provide it as soon as I am done.

**Comment by ram992** [comment]
Mean While can you suggest any patches that I can try with ??

**Comment by siralam** [comment]
Sorry that with current information I really can't come up with any idea that I think might be helpful.

If I encounter such situation, I will try to pin-point the problem first, like  
- Ensure the image has been downloaded;  
- Ensure the ImageView has loaded the image;
- Look at whether the ImageView is correctly measured

**Comment by ram992** [comment]
Hello @siralam,
While building the sample app I have noticed that using this `holder.mViewPager.setOffscreenPageLimit(myAdapter.getListCount());`, I could not reproduce the blank view issue. And same thing worked in my app too. So I wanted to ask two questions, will there be any memory impact by retaining all my pages in the memory, and also is it important to use aspectRatio that is there in the xml `app:viewpagerAspectRatio="1.0"` ??

**Comment by siralam** [comment]
Yes, there will be memory impact. I do not suggest setting off screen limit to a large number.

And yes, aspect ratio is important if your images are loaded from the internet.

From what you have mentioned, most likely there are something messed up with the fragment's lifecycle. Fragment lifecycle in Viewpager is a bit unobvious, on the other hand notifyDatasetChanged can also cause unexpected behavior.

If you can reproduce the problem without setting off screen limit, please share with me the sample project, I am happy to help.

**Comment by ram992** [comment]
Hello @siralam, Sorry to distrub again, I want to control the speed with which the pages in the viewPager change. Currently the animation is too fast, I saw another library called AutoScrollView Pager where i could do this but can't use it right now(cause of the blank view issue). Is there a way to do it in your library.

**Comment by siralam** [comment]
@ram992 Er.... Let me clarify, do you mean that your blank view issue exists also in another library named "AutoScrollView", but not anymore using my library?

Anyway, the flip page animation now is not configurable, because the animation was performed by `ViewPager` 's default select page action.  
If you google about this, [you can find a solution using reflection](https://stackoverflow.com/questions/10812009/change-viewpager-animation-duration-when-sliding-programmatically).

**Comment by ram992** [comment]
Yep, it exists in another Library with the same name(that is the reason I had to replace it urgently), and he is using some other RecyclerView Library from Jake Wharton. I don't know what the issue is, but I have tried numerous times to reproduce it in your lib but could not get it by setting the `setOffscreenPageLimit()` to that of my list, currently it's looking good and I am on a bit tight schedule.
And thanks for the link will try this out.
